### PR TITLE
fix typo DisclosureButtion -> DisclosureButton

### PR DESCRIPTION
**Pull request by Vanley**
### WHY are these changes introduced?

Two locations containing typo `DisclosureButtion`:
- polaris-react/src/components/Button/README.md 
- polaris.shopify.com/src/pages/generated-examples/Button-14.jsx 

### WHAT is this pull request doing?

Correcting typo in `DisclosureButtion` to `DisclosureButton`

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Comment by ghost** [comment]
👋 Thanks for opening your first pull request. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris/blob/master/.github/CONTRIBUTING.md).

**Review by martenbjork — APPROVED** [review]

**Comment by martenbjork** [comment]
Thanks for the fix! Now this just needs a changeset:

https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#adding-a-changeset

**Comment by alex-page** [comment]
@Vanley I think this PR might be out of date as it's referencing a directory and example that has been renamed. If you get the latest from `main` and want to make this change that would be ace. Thanks!

**Comment by Vanley** [comment]
> @Vanley I think this PR might be out of date as it's referencing a directory and example that has been renamed. If you get the latest from `main` and want to make this change that would be ace. Thanks!

@alex-page Cheers for the comment. Updated.

File: polaris-react/src/components/Button/README.md [hunk 1]
```diff
@@ -209,7 +209,7 @@ function PressedButton() {
 Use to indicate that more content can be disclosed on click, like text in a collapsible.
 
 ```jsx
-function DisclosureButtion() {
+function DisclosureButton() {
```
Comment (Vanley) [inline polaris-react/src/components/Button/README.md]:
```suggestion
function DisclosureButtion() {
```

**Review by alex-page — APPROVED** [review]

**Pull request merged**